pkg/cmd/historical_logs.go

```diff
@@ -107,40 +127,321 @@ func (o *LogParameters) Execute(streams genericclioptions.IOStreams) error {
 
 	ApiUrl := "http://log-exploration-api-route-openshift-logging.apps." + clusterName + "/logs/filter"
 
-	req, err := http.NewRequest("GET", ApiUrl, payload)
+	if len(o.Tail) > 0 {
+		tail, err := strconv.Atoi(o.Tail[0 : len(o.Tail)-1]) //extract numeric value. For example, extract 50 from 50s or 10 from 10m
+		if err != nil {
+			return fmt.Errorf("an invalid \"tail\" value was entered: %v", err)
+		}
 
-	if err != nil {
-		return fmt.Errorf("Request Failed: %v", err)
+		timeUnit := o.Tail[len(o.Tail)-1] //Last character (time unit) is 's'(seconds),'m'(minutes),'h'(hours),'d'(days)
+		endTime := time.Now().UTC()
+		var startTime time.Time
+
+		switch timeUnit {
+		case 's':
+			startTime = endTime.Add(time.Duration(-tail) * time.Second).UTC()
+		case 'm':
+			startTime = endTime.Add(time.Duration(-tail) * time.Minute).UTC()
+		case 'h':
+			startTime = endTime.Add(time.Duration(-tail) * time.Hour).UTC()
+		case 'd':
+			startTime = endTime.Add(time.Duration(-tail) * time.Hour * 24)
+		default:
+			return fmt.Errorf("invalid time unit entered in \"tail\". please enter s, m, h, or d as time unit")
+		}
+
+		o.StartTime = startTime.UTC().Format(time.RFC3339Nano)
+		o.EndTime = endTime.UTC().Format(time.RFC3339Nano)
 	}
 
-	client := &http.Client{}
-	res, err := client.Do(req)
-	if err != nil {
-		return fmt.Errorf("Response Error: %v", err)
+	var podList []string
+
+	if len(o.Deployment) > 0 {
+		err = GetDeploymentPodsList(clientset, o.Deployment, o.Namespace, &podList)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(o.DaemonSet) > 0 {
+		err = GetDaemonSetPodsList(clientset, o.DaemonSet, o.Namespace, &podList)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(o.StatefulSet) > 0 {
+		err = GetStatefulSetPodsList(clientset, o.StatefulSet, o.Namespace, &podList)
+		if err != nil {
+			return err
+		}
 	}
 
-	responseBody, err := ioutil.ReadAll(res.Body)
+	var logList []string
+	if len(podList) > 0 {
+		for _, pod := range podList {
+
+			o.Podname = pod
+			response, err := o.makeHttpRequest(ApiUrl)
+
+			if err != nil {
+				return err
+			}
+
+			responseBody, err := ioutil.ReadAll(response.Body)
+			if err != nil {
+				return fmt.Errorf("failed to read response: %v", err)
+			}
+
+			err = getLogs(responseBody, &logList, o.Prefix)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = printLogs(logList, streams, o.Limit)
+		if err != nil {
+			return err
+		}
+
+	} else {
+
+		response, err := o.makeHttpRequest(ApiUrl)
```

Comment:
@AkshayBhansali18 why do we need to create the target URL manually when we can send the query parameters very easily? The issue with the current approach is that in future if the no. of parameters will increase you would always need to extend this method. You can take a reference from here about [adding query parameters in a get request](https://github.com/ViaQ/log-exploration-api/blob/main/pkg/controllers/logs/controller_test.go#L126).